src/main/scala/com/lambtors/poker_api/module/poker/application/create/PokerGameCreator.scala

```diff
@@ -0,0 +1,16 @@
+package com.lambtors.poker_api.module.poker.application.create
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import com.lambtors.poker_api.module.poker.domain.PokerGameRepository
+import com.lambtors.poker_api.module.poker.domain.error.PokerGameAlreadyExisting
+import com.lambtors.poker_api.module.poker.domain.model.{AmountOfPlayers, GameId, PokerGame}
+
+final class PokerGameCreator(pokerGameRepository: PokerGameRepository)(implicit ec: ExecutionContext) {
+  def create(amountOfPlayers: AmountOfPlayers, gameId: GameId): Future[Unit] = {
+    pokerGameRepository
+      .search(gameId)
```

Comment:
we still have to flatmap the result, so I think it makes more sense to flatmap directly